### ISSUE TITLE
Player Heads picked up by mobs create duplicates

### ISSUE THREAD

**Issue by Skizzles** [BG-Software-LLC/WildStacker#578]
### Minecraft's Version

1.19.1 purpur dev 1749

### Plugin's Version

Dev 331

### Describe the bug

If a player head is on the ground and a stacked mob picks it up then it sometimes allows a duplicate as it gets picked up and stays on the ground.

### To Reproduce

https://screenshots.skizzles.me/wAnA3/luyoYaHa27.mp4

**Comment by OmerBenGera** [comment]
Check latest dev build, let me know if it fixes the issue.
I fixed another duplication glitch that sounds the same, so it might fix this one as well

**Comment by Skizzles** [comment]
> Check latest dev build, let me know if it fixes the issue. I fixed another duplication glitch that sounds the same, so it might fix this one as well

Seems fixed, but I dont see the item dropping at all now.

**Comment by OmerBenGera** [comment]
> > Check latest dev build, let me know if it fixes the issue. I fixed another duplication glitch that sounds the same, so it might fix this one as well
> 
> Seems fixed, but I dont see the item dropping at all now.

Wdym?

**Comment by Skizzles** [comment]
> > > Check latest dev build, let me know if it fixes the issue. I fixed another duplication glitch that sounds the same, so it might fix this one as well
> > 
> > 
> > Seems fixed, but I dont see the item dropping at all now.
> 
> Wdym?

The player head on the mob is not given back, it just poofs

**Comment by OmerBenGera** [comment]
> > > > Check latest dev build, let me know if it fixes the issue. I fixed another duplication glitch that sounds the same, so it might fix this one as well
> > > 
> > > 
> > > Seems fixed, but I dont see the item dropping at all now.
> > 
> > 
> > Wdym?
> 
> The player head on the mob is not given back, it just poofs

Is it another plugin that handles player heads?
If so, mind giving me a link so I can try replicating it myself?

**Comment by Skizzles** [comment]
> > > > > Check latest dev build, let me know if it fixes the issue. I fixed another duplication glitch that sounds the same, so it might fix this one as well
> > > > 
> > > > 
> > > > Seems fixed, but I dont see the item dropping at all now.
> > > 
> > > 
> > > Wdym?
> > 
> > 
> > The player head on the mob is not given back, it just poofs
> 
> Is it another plugin that handles player heads? If so, mind giving me a link so I can try replicating it myself?

I tried this with player heads from Slimefun and also HeadDatabase, but also with just vanilla skeleton/zombie heads

**Comment by OmerBenGera** [comment]
> > > > > > Check latest dev build, let me know if it fixes the issue. I fixed another duplication glitch that sounds the same, so it might fix this one as well
> > > > > 
> > > > > 
> > > > > Seems fixed, but I dont see the item dropping at all now.
> > > > 
> > > > 
> > > > Wdym?
> > > 
> > > 
> > > The player head on the mob is not given back, it just poofs
> > 
> > 
> > Is it another plugin that handles player heads? If so, mind giving me a link so I can try replicating it myself?
> 
> I tried this with player heads from Slimefun and also HeadDatabase, but also with just vanilla skeleton/zombie heads

How do you try to get skeleton/zombie heads to be dropped? using a charged creeper?

**Comment by OmerBenGera** [comment]
@Skizzles I fixed a bug with filters of loot not checked correctly, which caused some loot to not drop when it should have been.
Update to latest dev build, let me know if it works now

**Comment by Skizzles** [comment]
> @Skizzles I fixed a bug with filters of loot not checked correctly, which caused some loot to not drop when it should have been. Update to latest dev build, let me know if it works now

managed to get it to dupe again... seems really odd behavior

**Comment by OmerBenGera** [comment]
> > @Skizzles I fixed a bug with filters of loot not checked correctly, which caused some loot to not drop when it should have been. Update to latest dev build, let me know if it works now
> 
> managed to get it to dupe again... seems really odd behavior

How to replicate the issue? Also, why would it only happen with skulls?
Are you sure there's no other plugins that just drops an additional skull?

**Comment by Skizzles** [comment]
> > > @Skizzles I fixed a bug with filters of loot not checked correctly, which caused some loot to not drop when it should have been. Update to latest dev build, let me know if it works now
> > 
> > 
> > managed to get it to dupe again... seems really odd behavior
> 
> How to replicate the issue? Also, why would it only happen with skulls? Are you sure there's no other plugins that just drops an additional skull?

Yes as I tested this with just wildstacker installed, the mob picks up the skull and puts it on its head and dupes the skull when killed. This specifically is happening to me with wither skeletons so maybe related to the fact that they already drop skulls?

**Comment by OmerBenGera** [comment]
> > > > @Skizzles I fixed a bug with filters of loot not checked correctly, which caused some loot to not drop when it should have been. Update to latest dev build, let me know if it works now
> > > 
> > > 
> > > managed to get it to dupe again... seems really odd behavior
> > 
> > 
> > How to replicate the issue? Also, why would it only happen with skulls? Are you sure there's no other plugins that just drops an additional skull?
> 
> Yes as I tested this with just wildstacker installed, the mob picks up the skull and puts it on its head and dupes the skull when killed. This specifically is happening to me with wither skeletons so maybe related to the fact that they already drop skulls?

Take a video of the issue

**Comment by Skizzles** [comment]
![image](https://user-images.githubusercontent.com/30563757/187145491-33c6786a-9974-4d74-a933-5608ea04ad56.png)
https://screenshots.skizzles.me/wAnA3/raruMaBe69.mp4

**Comment by OmerBenGera** [comment]
> ![image](https://user-images.githubusercontent.com/30563757/187145491-33c6786a-9974-4d74-a933-5608ea04ad56.png)
> 
> Video https://screenshots.skizzles.me/wAnA3/raruMaBe69.mp4

Just to make sure I understand the issue correctly,
the intended behavior that the skull will disappear after one entity dies from the stack, correct?

**Comment by Skizzles** [comment]
> > ![image](https://user-images.githubusercontent.com/30563757/187145491-33c6786a-9974-4d74-a933-5608ea04ad56.png)
> > Video https://screenshots.skizzles.me/wAnA3/raruMaBe69.mp4
> 
> Just to make sure I understand the issue correctly, the intended behavior that the skull will disappear after one entity dies from the stack, correct?

Yes, that the skull would basically be dropped and not continue into the next stack. Currently it drops but also continues into the stack allowing the skull to dupe.

**Comment by OmerBenGera** [comment]
Seems to be caused by #596
